File: packages/constants/src/webhook.ts
/**
 * Webhook相关常量
 */

/** Webhook目标类型 */
export const WEBHOOK_TARGET_TYPE = {
  /** 门店 */
  STORE: 'store',
  /** 供应商 */
  SUPPLIER: 'supplier',
} as const;

/** Webhook事件类型 */
export const WEBHOOK_EVENT_TYPE = {
  /** 订单创建 */
  ORDER_CREATED: 'order_created',
  /** 订单确认 */
  ORDER_CONFIRMED: 'order_confirmed',
  /** 订单配送中 */
  ORDER_DELIVERING: 'order_delivering',
  /** 订单完成 */
  ORDER_COMPLETED: 'order_completed',
  /** 订单取消 */
  ORDER_CANCELLED: 'order_cancelled',
  /** 订单恢复 */
  ORDER_RESTORED: 'order_restored',
} as const;

/** Webhook事件类型文本 */
export const WEBHOOK_EVENT_TYPE_TEXT: Record<string, string> = {
  [WEBHOOK_EVENT_TYPE.ORDER_CREATED]: '新订单',
  [WEBHOOK_EVENT_TYPE.ORDER_CONFIRMED]: '订单确认',
  [WEBHOOK_EVENT_TYPE.ORDER_DELIVERING]: '开始配送',
  [WEBHOOK_EVENT_TYPE.ORDER_COMPLETED]: '订单完成',
  [WEBHOOK_EVENT_TYPE.ORDER_CANCELLED]: '订单取消',
  [WEBHOOK_EVENT_TYPE.ORDER_RESTORED]: '订单恢复',
};

/** Webhook推送状态 */
export const WEBHOOK_STATUS = {
  /** 待推送 */
  PENDING: 'pending',
  /** 推送成功 */
  SUCCESS: 'success',
  /** 推送失败 */
  FAILED: 'failed',
} as const;

/** Webhook推送状态文本 */
export const WEBHOOK_STATUS_TEXT: Record<string, string> = {
  [WEBHOOK_STATUS.PENDING]: '待推送',
  [WEBHOOK_STATUS.SUCCESS]: '成功',
  [WEBHOOK_STATUS.FAILED]: '失败',
};

/** Webhook推送状态颜色 */
export const WEBHOOK_STATUS_COLOR: Record<string, string> = {
  [WEBHOOK_STATUS.PENDING]: '#f59e0b', // 橙色
  [WEBHOOK_STATUS.SUCCESS]: '#22c55e', // 绿色
  [WEBHOOK_STATUS.FAILED]: '#ef4444', // 红色
};

/** 默认Webhook事件（供应商） */
export const DEFAULT_SUPPLIER_WEBHOOK_EVENTS = [
  WEBHOOK_EVENT_TYPE.ORDER_CREATED,
  WEBHOOK_EVENT_TYPE.ORDER_CANCELLED,
] as const;

/** 默认Webhook事件（门店） */
export const DEFAULT_STORE_WEBHOOK_EVENTS = [
  WEBHOOK_EVENT_TYPE.ORDER_CONFIRMED,
  WEBHOOK_EVENT_TYPE.ORDER_DELIVERING,
  WEBHOOK_EVENT_TYPE.ORDER_COMPLETED,
] as const;

File: apps/backend/types/webhook.go
package types

import "time"

// WebhookEvent Webhook事件枚举
type WebhookEvent string

const (
	WebhookNewOrder       WebhookEvent = "new_order"
	WebhookOrderConfirmed WebhookEvent = "order_confirmed"
	WebhookOrderDelivered WebhookEvent = "order_delivered"
	WebhookOrderCompleted WebhookEvent = "order_completed"
	WebhookOrderCancelled WebhookEvent = "order_cancelled"
	WebhookPriceUpdated   WebhookEvent = "price_updated"
	WebhookStockChanged   WebhookEvent = "stock_changed"
)

// WebhookTargetType Webhook目标类型
type WebhookTargetType string

const (
	WebhookTargetStore    WebhookTargetType = "store"
	WebhookTargetSupplier WebhookTargetType = "supplier"
)

// WebhookStatus Webhook状态
type WebhookStatus string

const (
	WebhookPending WebhookStatus = "pending"
	WebhookSuccess WebhookStatus = "success"
	WebhookFailed  WebhookStatus = "failed"
)

// WebhookPayload Webhook推送载荷
type WebhookPayload struct {
	Event     WebhookEvent `json:"event"`
	Timestamp int64        `json:"timestamp"`
	Data      interface{}  `json:"data"`
	Signature string       `json:"signature,omitempty"`
}

// WebhookLogInfo Webhook推送日志信息
type WebhookLogInfo struct {
	ID            uint64            `json:"id"`
	TargetType    WebhookTargetType `json:"targetType"`
	TargetID      uint64            `json:"targetId"`
	Event         WebhookEvent      `json:"event"`
	WebhookURL    string            `json:"webhookUrl"`
	RequestBody   string            `json:"requestBody"`
	ResponseCode  int               `json:"responseCode,omitempty"`
	ResponseBody  string            `json:"responseBody,omitempty"`
	Status        WebhookStatus     `json:"status"`
	RetryCount    int               `json:"retryCount"`
	NextRetryTime *time.Time        `json:"nextRetryTime,omitempty"`
	ErrorMessage  string            `json:"errorMessage,omitempty"`
	CreatedAt     time.Time         `json:"createdAt"`
	UpdatedAt     time.Time         `json:"updatedAt"`
}

// WebhookQueryParams Webhook日志查询参数
type WebhookQueryParams struct {
	PaginationQuery
	TargetType WebhookTargetType `json:"targetType" query:"targetType"`
	TargetID   uint64            `json:"targetId" query:"targetId"`
	Event      WebhookEvent      `json:"event" query:"event"`
	Status     WebhookStatus     `json:"status" query:"status"`
	StartDate  string            `json:"startDate" query:"startDate"`
	EndDate    string            `json:"endDate" query:"endDate"`
}

// WebhookConfigInfo Webhook配置信息
type WebhookConfigInfo struct {
	WebhookURL     string         `json:"webhookUrl"`
	WebhookEnabled bool           `json:"webhookEnabled"`
	WebhookEvents  []WebhookEvent `json:"webhookEvents"`
}

// UpdateWebhookConfigRequest 更新Webhook配置请求
type UpdateWebhookConfigRequest struct {
	WebhookURL     string         `json:"webhookUrl" validate:"url"`
	WebhookEnabled bool           `json:"webhookEnabled"`
	WebhookEvents  []WebhookEvent `json:"webhookEvents"`
}

// OrderWebhookData 订单Webhook数据
type OrderWebhookData struct {
	OrderNo        string      `json:"orderNo"`
	OrderID        uint64      `json:"orderId"`
	Status         OrderStatus `json:"status"`
	TotalAmount    float64     `json:"totalAmount"`
	ItemCount      int         `json:"itemCount"`
	StoreName      string      `json:"storeName,omitempty"`
	SupplierName   string      `json:"supplierName,omitempty"`
	DeliveryDate   string      `json:"deliveryDate,omitempty"`
	Remark         string      `json:"remark,omitempty"`
}

// PriceWebhookData 价格变动Webhook数据
type PriceWebhookData struct {
	MaterialSkuID uint64  `json:"materialSkuId"`
	MaterialName  string  `json:"materialName"`
	Brand         string  `json:"brand"`
	Spec          string  `json:"spec"`
	OldPrice      float64 `json:"oldPrice"`
	NewPrice      float64 `json:"newPrice"`
	SupplierID    uint64  `json:"supplierId"`
	SupplierName  string  `json:"supplierName"`
}

// StockWebhookData 库存变动Webhook数据
type StockWebhookData struct {
	MaterialSkuID uint64      `json:"materialSkuId"`
	MaterialName  string      `json:"materialName"`
	Brand         string      `json:"brand"`
	Spec          string      `json:"spec"`
	OldStatus     StockStatus `json:"oldStatus"`
	NewStatus     StockStatus `json:"newStatus"`
	SupplierID    uint64      `json:"supplierId"`
	SupplierName  string      `json:"supplierName"`
}
